### ISSUE TITLE
[Feature]: More flexible, easier way to specify Optimizer other than Adam

### ISSUE THREAD

**Issue by yonatank93** [AUTODIAL/AutoREC#26]
## Motivation

In the current version, the optimizer of choice is Adam, and so we can pass in "adam" string to the optimizer option in DDQN_ECM agent, and it will create it with the default parameters (except learning rate). If we want to use a different optimizer (or even just use non-default Adam parameters), we need to instantiate the optimizer first, then pass it to DDQN_ECM.
I was wondering if we can make this process easier: using different optimizer or optimizer's parameters.

## Proposed solution

We can overhaul how the optimizer is specified. We still accept the optimizer instance. However, we can also accept a dictionary containing configurations on how the optimizer is instantiated.
There are several benefits from doing this. We can:
1. easily use different TensorFlow built-in optimizer with some custom parameters.
2. define different optimizer configuration from the configuration.
3. have shareable text metadata.

## Alternatives considered

I think TensorFlow has some functionalities to help us with this suggestion.
